source/common/network/address_impl.cc

```diff
@@ -173,10 +173,10 @@ std::string Ipv4Instance::sockaddrToString(const sockaddr_in& addr) {
 }
 
 absl::uint128 Ipv6Instance::Ipv6Helper::address() const {
-  absl::uint128 result{0};
   static_assert(sizeof(absl::uint128) == 16, "The size of asbl::uint128 is not 16.");
-  memcpy(static_cast<void*>(&result), static_cast<const void*>(&address_.sin6_addr.s6_addr),
-         sizeof(absl::uint128));
+  unsigned __int128 ipv6_addr;
```

Comment:
is this portable? Why don't you just declare this as absl::uint128 and skip the copy on line 179?

Reply:
Well ^^, that was the reason for placing the check (ifdef) in the first place.  Unfortunately, I got an error when trying to initialize ```absl::uint128``` with ```address_.sin6_addr.s6_addr``` since there is no constructor that accept a unsigned char*. So, maybe it's better if I just leave the original memcpy, do you agree?

Reply:
I'm not understanding why you need to construct anything. Can you just write:
```
absl::uint128 result;
static_assert(sizeof(absl::uint128) == 16, "The size of asbl::uint128 is not 16.");
safe_memcpy(&result, reinterpret_cast<const absl::uint128*>&(address_.sin6_addr.s6_addr));
return result;
```
what error do you get when you try that? Note that you don't have to build everything locally: you can just type:
```
bazel build source/common/network:address_lib
```